src/libs/fileDownload/index.android.js

```diff
@@ -7,6 +7,12 @@ import * as FileUtils from './FileUtils';
  * @returns {Promise<Boolean>}
  */
 function hasAndroidPermission() {
+    // On Android API Level 33 and above, these permissions do nothing and always return 'never_ask_again'
+    // More info here: https://stackoverflow.com/a/74296799
```

Comment:
NAB: I think your explanation is enough here and you might not need the S.O link